### PR TITLE
Standardize script field's rejection error

### PR DESCRIPTION
**Pull request by nik9000**
This standardizes the exception type thrown when attempting to build a
query on a script field that it doesn't support. It also replaces the
field type `[script]` in the error message with whatever the runtime type
is.

### CODE REVIEW

**Review by javanna — APPROVED** [review]
left one comment, LGTM otherwise

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java [hunk 1]
```diff
@@ -49,6 +60,90 @@ public final boolean isAggregatable() {
...
+
+    public Query regexpQuery(
+        String value,
+        int flags,
+        int maxDeterminizedStates,
+        MultiTermQuery.RewriteMethod method,
+        QueryShardContext context
+    ) {
+        throw new IllegalArgumentException(
+            "Can only use regexp queries on keyword and text fields - not on [" + name() + "] which is of type [" + runtimeType() + "]"
```
[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java:117]: :+1:

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java [hunk 2]
```diff
@@ -49,6 +60,71 @@ public final boolean isAggregatable() {
...
+    public Query phrasePrefixQuery(TokenStream stream, int slop, int maxExpansions) throws IOException {
+        throw new IllegalArgumentException(unsupported("phrase prefix", "text"));
+    }
+
+    public SpanQuery spanPrefixQuery(String value, SpanMultiTermQueryWrapper.SpanRewriteMethod method, QueryShardContext context) {
+        throw new IllegalArgumentException(unsupported("span prefix", "text"));
+    }
+
+    private String unsupported(String query, String supported) {
+        String thisField = "[" + name() + "] which is of type [script] with runtime_type [" + runtimeType() + "]";
```
Comment (javanna) [inline x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java:124]: don't hardcode script :) we are going to rename it and if you use the existing constant renaming will be easier

**Review by javanna — APPROVED** [review]
LGTM besides the minor comment I left

**Pull request merged**